chainer/links/caffe/caffe_function.py

```diff
@@ -435,3 +484,135 @@ def __init__(self, caffe_func, name):
 
     def __call__(self, *xs):
         return self.caffe_func[self.name](*xs)
+
+
+class _EltwiseFunction(object):
+    def __init__(self, operation, coeffs=None):
+        if coeffs is not None:
+            assert len(coeffs) > 0
+        self.operation = operation
+        self.coeffs = coeffs
+
+    def __call__(self, *xs):
+        operation = self.operation
+
+        if operation == 0:      # PROD
+            return reduce(lambda x, y: x * y, xs),
+
+        elif operation == 1:    # SUM
+            coeffs = self.coeffs
+            if coeffs is not None:
+                assert len(xs) == len(coeffs)
+                xs2 = [x * coeff for x, coeff in zip(xs, coeffs)]
+            else:
+                xs2 = xs
+            return reduce(lambda x, y: x + y, xs2),
+
+        elif operation == 2:    # MAX
+            return reduce(lambda x, y: functions.maximum(x, y), xs),
+
+        else:
+            raise RuntimeError('Invalid EltwiseParameter.EltwiseOp value.')
+
+
+class _ScaleFunction(object):
+    def __init__(self, axis):
+        self.axis = axis
+
+    def __call__(self, x, y):
+        x_shape = x.data.shape
+        y_shape = y.data.shape
+        axis = self.axis
+        assert x_shape[axis:axis + len(y_shape)] == y_shape
+        y1_shape = tuple([1] * axis + list(y_shape) +
+                         [1] * (len(x_shape) - axis - len(y_shape)))
+        y1 = functions.reshape(y, y1_shape)
+        y2 = functions.broadcast_to(y1, x_shape)
+        return x * y2,
+
+
+class _ScaleChain(link.Chain):
+    def __init__(self, axis=1, W_blob=None, bias_blob=None):
+        super(_ScaleChain, self).__init__()
+
+        # Add W parameter if given.
+        if W_blob is not None:
+            W_shape = tuple(W_blob.shape.dim)
+            self.add_param('W', W_shape)
+            self.W.data.ravel()[:] = W_blob.data
```

Comment:
Caffe blob's data are flattened one-dimensional lists, so `W_blob.data` needs its dimension changed. I suppose that `self.W.data[...] = W_blob.data` does not work in this case.